Clinical-Genomics/scout
# Populate case group manual assessments after collecting case group evaluated

An attempt to speed up the variants page loading. Fix #2410

Current situation, simplified:

assuming casex is in the same group with casej:

```
for variant in casex vars: --> one variants page from casex (for instance 50 vars)
    for case in case_group:
        # found casej in group
        # Look inside all variants of casej to see if there is same variant as variant (casex)
        # See if this variant has assessments
```

Proposed change:
- Before looping thru the case variants, retrieve all evaluated cases for other cases in the group (well established function used all over the place) and create a dictionary like this:
```
{
 variant_x_simple_id = [list of all evaluated variants in case groups with that simple_id],
 variant_y_simple_id = [list of all evaluated variants in case groups with that simple_id],
..
}
```
- Then when looping thru the case variants, create assessment list based on the above dict.

This approach looks simpler, reuses available functions and should entail a limited number of queries.


**How to test**:
1. Check loading variants page of a case contained in a case group and see if it's any faster
1. Make sure that evaluated variants present also in other cases of the same group are still visualized for the case

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
